docs/properties_and_constraints.md

```diff
@@ -0,0 +1,109 @@
+	"value": <property-value here>
+}
+```
+
+The following is an example using each of the property types:
+```
+{
+	"name": "stringProperty",       /* type is omitted to demonstrate that OpenHorizon will interpret this property as a string type */
+	"value": "string-value"
+},
+{
+	"name": "intProperty",          /* type is omitted to demonstrate that OpenHorizon will interpret this property as an int type */
+	"value": 10
+},
+{
+	"name": "booleanProperty",      /* type is omitted to demonstrate that OpenHorizon will interpret this property as a string type */
+	"value": true
+},
+{
+	"name": "floatProperty",        /* type is omitted to demonstrate that OpenHorizon will interpret this property as a string type */
+	"value": 3.4
+},
+{
+	"name": "versionProperty",      /* type is specified to demonstrate that OpenHorizon would otherwise interpret this property as a string */
+	"type": "version",
+	"value": "1.2.7"
+},
+{
+	"name": "losProperty",          /* type is specified to demonstrate that OpenHorizon would otherwise interpret this property as a string */
+	"type": "list of strings",
+	"value": "value1,value2"
+}
+```
+
+## Constraints
+
+Constraints are the other half of the policy expression language in OpenHorizon.
+Think of constraints as selection predicates used to select nodes based on the properties defined and set on those nodes.
+Constraints are most commonly found in deployment policies but can be specified on nodes, service and model policy.
+
+Constraints are specified using a simple text based language.
+The lexical parser used to interpret the language is described by an ebnf in a function called getLexer() in [the code](../externalpolicy/text_language/text_language.go).
+The language allows property name references and their expected values to be strung together with boolean operators `AND` and `OR` into boolean expressions.
+The more golang-like boolean operators (`&&` and `||`) are also supported.
+Parentheses are supported in order to create evaluation precedence.
+The boolean operator `NOT` is not currently supported in constraint expressions.
+
+When a constraint expression is evaluated against a list of properties, the result will be either true or false.
+True means that the constraint is compatible with the property list, false means it is not compatible.
+
+For example, using the example properties defined in the properties section above, the constraint:
+`"booleanProperty = true AND floatProperty < 1.0"` will evaluate to false.
+
+Each property type has operators that can be used to evaluate property values:
+* `string` - the operators `==` or `=` denote equals to and `!=` denotes not equal to.
+* `int` - supports the operators `==, <, >, <=, >=, =, !=`.
+* `boolean` - supports `==, =`
+* `float` - supports the operators `==, <, >, <=, >=, =, !=`.
+* `version` - supports `==, =, in` where in is used to indicate that a version is within a given range, e.g. any version 1 service: "(1.0.0,2.0.0]".
```

Comment:
any version 1 service?

Reply:
yes, isn't that the version range for any version 1 service?

Reply:
You are right. I was not sure what you were talking about. Now I understand.

Reply:
fixed